azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineScaleSetVMImpl.java

```diff
@@ -380,21 +380,30 @@ public VirtualMachineInstanceView instanceView() {
 
     @Override
     public VirtualMachineInstanceView refreshInstanceView() {
-        VirtualMachineScaleSetVMInstanceViewInner instanceViewInner = this.client.getInstanceView(this.parent().resourceGroupName(),
-                this.parent().name(),
-                this.instanceId());
-        if (instanceViewInner != null) {
-            this.virtualMachineInstanceView = new VirtualMachineInstanceView()
-                    .withBootDiagnostics(instanceViewInner.bootDiagnostics())
-                    .withDisks(instanceViewInner.disks())
-                    .withExtensions(instanceViewInner.extensions())
-                    .withPlatformFaultDomain(instanceViewInner.platformFaultDomain())
-                    .withPlatformUpdateDomain(instanceViewInner.platformUpdateDomain())
-                    .withRdpThumbPrint(instanceViewInner.rdpThumbPrint())
-                    .withStatuses(instanceViewInner.statuses())
-                    .withVmAgent(instanceViewInner.vmAgent());
-        }
-        return this.virtualMachineInstanceView;
+        return refreshInstanceViewAsync().toBlocking().last();
```

Comment:
(y)
Cool.. Since this is refresh call, it is good to pick last element. 
Will **.first()** and **.last()** make any difference here?

If I call refreshInstanceViewAsync(), it will **only emit one item**. Correct me if I am wrong.

Reply:
That's right, in this case refreshInstanceViewAsync will only emit one item so .first() or .last() won't really make a difference.